.github/workflows/release.yml

```diff
@@ -47,3 +49,18 @@ jobs:
             ./dist/*.whl
           prerelease: ${{ contains(github.ref, '-rc') }}
           generate_release_notes: ${{ !contains(github.ref, '-rc') }}
```

Comment:
Can we also include the URL path to the newly published PyPI package here?

See docs in [`softprops/action-gh-release`](https://github.com/softprops/action-gh-release?tab=readme-ov-file#-customizing)

Note we might want to reorder these steps so the PyPI publish happens before the GitHub release.

Reply:
This might be more trouble than it is worth until the GH Actions supports this as an output, see pypa/gh-action-pypi-publish/issues/97

We could try and construct the URL, which will always be known -- but the problem is the version format difference between the git tag `YYYY.0M.R` (0-padded month) and the Python package `YYYY.M.R` (non-0-padded month).

If this seems simple enough, go ahead and implement, otherwise we can wait.

Reply:
@thekaveman Here's where I'm currently at with this:

I started an attempt with [these commits](https://github.com/cal-itp/littlepay/pull/31/files/53483f5b3e67676cb3ef96c478400577e362fd26..76878f1b4fed9aded513c4c343b06edc0a54e0d4) -- I just wanted to see if I could even get the URL to show up in the release notes. These changes don't handle the version format difference between the 0-padded month and non-0-padded month.

With that rough draft, I tried to test it with [a manual run](https://github.com/cal-itp/littlepay/actions/runs/8254409896), but the "Coverage comment" step fails because it [doesn't support `workflow_dispatch` for Python 3.11](https://github.com/cal-itp/littlepay/actions/runs/8254409896/job/22578554741#step:7:26) (🤷) so I disabled them in https://github.com/cal-itp/littlepay/pull/31/commits/e63a9d7703d749d05d1e7eba54f2d2639833b00c. I then tried [another manual run](https://github.com/cal-itp/littlepay/actions/runs/8254443582) and found I needed to remove the environment protection in order to release from this branch. After several more attempts, I'm at a point now where I was hoping it'd create a pre-release with the test PyPI URl in the release notes, but now the runner [fails on my `bash` commands](https://github.com/cal-itp/littlepay/actions/runs/8254826399/job/22579868473#step:9:62) that attempt to parse out the version and construct the URL.

Reply:
For the bash string manipulation, I assume you're looking at something like this? https://tldp.org/LDP/abs/html/string-manipulation.html

Have you tried the `${string#substring}` syntax?

> Deletes shortest match of `$substring` from front of `$string`.

So I think that would be: `${output:littlepay }`

Reply:
I was referencing the first StackOverflow that my search results gave me (https://stackoverflow.com/questions/13210880/replace-one-substring-for-another-string-in-shell-script)

It works in my devcontainer bash console as well:
```
$ output=$(littlepay -v)
$ version=${output/littlepay/ }
$ echo $version
2023.11.2.dev18+gc2d8b87.d20240213
```

so I feel like it's something specific to the GitHub runner 😕 

The [error messages](https://github.com/cal-itp/littlepay/actions/runs/8254826399/job/22579868473#step:9:62) imply to me that it was able to get the version number out (it shows the version number without `littlepay ` in front...), but the messages are not helpful in what exactly is failing.

```
Error: Unable to process file command 'output' successfully.
Error: Invalid format '0.1.dev1+gbe36cef'
```

At this point, I'm resorting to trying things out in my `github-experiments` repo.

Reply:
🤦 My syntax was wrong

The error messages were talking about a failure to set the GitHub output. Let's see how much further I can get now 🤞 

https://stackoverflow.com/questions/74137120/how-to-fix-or-avoid-error-unable-to-process-file-command-output-successfully

Reply:
The last problem to overcome was that `littlepay -v` was not only printing the version, but also creating a new config file since this workflow doesn't point it at existing file. This caused the output of `littlepay -v` to be something like this:
```
Creating config file: /home/runner/.littlepay/config.yaml 
0.1.dev1g9bfec47 
```
which leads to the error message about a multi-line output variable not being set correctly.

I resolved this by just calling `littlepay` before doing anything else.